From 96306747f46f6c9f770bbe5b8adf22121063f087 Mon Sep 17 00:00:00 2001
From: Alexandru Croitor <alexandru.croitor@qt.io>
Date: Wed, 20 Sep 2023 15:47:32 +0200
Subject: [PATCH] CMake: Fix zstd::libzstd_shared global target promotion issue

The zstd target might be implicitly created by
qt_find_package(WrapLibClang) in configure.cmake via the
LLVMConfig.cmake file, because it has a dependency on zstd.

Then we try to promote the target to global in the designer src
subdirectory, which fails.

Check if the target already exists, in which case we skip the global
promotion.

Similar to 381994598546131f262a3abac7fbcc325acfc7b5 in qtimageformats.

Amends a0ecd3b3f7855cba3f8e91aff6617039e641da62

Fixes: QTBUG-117145
Change-Id: I6e946f9fb8130d8e2513f164e99a1ac5dfb9c1ac
Reviewed-by: Alexey Edelev <alexey.edelev@qt.io>
Reviewed-by: Joerg Bornemann <joerg.bornemann@qt.io>
(cherry picked from commit 7a710de3fd5d667dbe4fd108a0954abdc6732fa8)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---

diff --git a/src/designer/src/lib/CMakeLists.txt b/src/designer/src/lib/CMakeLists.txt
index db51e77..edd7c7f 100644
--- a/src/designer/src/lib/CMakeLists.txt
+++ b/src/designer/src/lib/CMakeLists.txt
@@ -411,6 +411,9 @@
         ../../../shared/qtpropertybrowser
 )
 
+if(TARGET zstd::libzstd_shared)
+    qt_internal_disable_find_package_global_promotion(zstd::libzstd_shared)
+endif()
 if(NOT TARGET WrapZSTD::WrapZSTD)
     qt_find_package(WrapZSTD 1.3 PROVIDED_TARGETS WrapZSTD::WrapZSTD)
 endif()
